exercises/tom_clayton/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,114 @@
+#include "Shapesorter.h"
+#include <iostream>
+#include <algorithm>
+#include <vector>
+
+//ShapeSorter constructor
+ShapeSorter::ShapeSorter(std::vector<Shape> shpVect) {
+  m_shpVect = shpVect;
+  m_vecLen = shpVect.size();
+}
+
+// Typedef to be used to iterate through the vector
+typedef std::vector<Shape>::size_type VecSz;
+
+/** Print out all shapes of a given type, or an error message if no
+ *  shapes of that type are found
+ *  @param string type for type of string
+ */ 
+const void ShapeSorter::printShapesType(const std::string type) {
+  bool flag = false;
+  for(VecSz i = 0; i < getVecLen(); ++i) {
+    if(m_shpVect[i].getType().compare(type) == 0) {
+      m_shpVect[i].shapeToString();
+      flag = true;
+    }
+  }
+  if(!flag) {
+    std::cout << "No shapes found of type " << type << std::endl;
+  }
+}
+
+/** Print out all shapes of a given number of sides, or an error
+ *  message if no shapes of that type are found
+ *  @param int sides for no of sides
+ */ 
+const void ShapeSorter::printShapesSides(int sides) {
+  bool flag = false;

```

Comment:
`flag` isn't a very useful name as you don't know what this flag is for. Any variable should be clear what it is without having to look through all the code